### DISCUSSION TITLE
Sign Up with OAuth > Access Query Params from a DB Trigger function

### DISCUSSION THREAD

**Discussion by Metaphysics0** [supabase/supabase#10545] [Questions]
I'm adding sign up with OAuth (with spotify) for an app I'm building, and I want to ask for some initial information from the user upon sign up.

In the [docs](https://supabase.com/docs/reference/javascript/auth-signinwithoauth), it states that we can pass an optional `queryParams` object to the request payload.

Is there a way to access these query params with a Supabase DB trigger function (`on_auth_user_created` for example), to store in the profile upon creation?

In addition, I noticed in the `on_auth_user_created` trigger provided within user management starter example, I noticed we are accessing `new.raw_user_meta_data` like this:
```
begin
  insert into public.profiles (id, full_name, avatar_url)
  values (new.id, new.raw_user_meta_data->>'full_name', new.raw_user_meta_data->>'avatar_url');
  return new;
end;
```

I would like to do something like this (in pseudo code):
```
begin 
  ...
  values(new.query_params_from_sign_up ->>....
```

Thanks so much.

**Comment by GaryAustin1** [comment]
I don't think you can.

The parameters seemed to be added for dealing with an OAUTH provider.  https://github.com/supabase/gotrue-js/issues/279

Unless gotrue provides them to the database as a setting you would not be able to access, I don't see anything to indicated they do.

You should just be able to use the user meta data to pass in.  Then if you don't want it saved in the auth.users table and the jwt, modify the new value to not include your data.   This would have to occur on a before insert trigger on auth.users, if you want to just change the new object to be inserted.
